tabsie/server.R

```diff
@@ -255,18 +257,24 @@ shinyServer(
     output$freqTable <- renderTable({#validation done before this is called, no need to repeat
       if (!valAuth) return;#break processing of not authorized.
       pdata = getpData(input$filter, serverDataDic, serverData)
-      addmargins(table(pdata[,c(input$xVal,input$yVal)]))
-    })
+      if(input$xVal==input$yVal) {
```

Comment:
Right now, if two categorics are chosen where they are the same item, the default behavior had been to create a table which is all zeros except for the diagonal. Now this case is detected and a single-column frequency table is created.